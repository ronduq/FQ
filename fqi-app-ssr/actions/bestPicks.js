import { 
  SET_BEST_PICKS,
  SORT_BEST_PICKS,
  SET_INITIAL_BEST_PICKS_CRITERIA
} from '../constants/'

import { setContentError } from './content';
import dataFetcher from '../lib/dataFetcher';

import {
  getSelectedLocation,
  getSelectedProduce,
  getSelectedRetailers,
  getBestPicksSortBy
} from '../selectors/index';

export const generateBestPicksAndFetch = (dispatch, state, location, criteria) => {
  const url = `${process.env.BASE_URL}/api/v1/${location}/produce/picks`;
  const produceCodes = getSelectedProduce(state);
  const retailerCodes = getSelectedRetailers(state);
  const sortBy = criteria || getBestPicksSortBy(state);;

  const produceQuery = { 
    produceCodes, 
    retailerCodes, 
    sortBy,
  };    
 
  return new Promise(resolve => {
      dispatch(fetchBestPicksData(resolve, url, produceQuery));
  });
}

export const fetchBestPicksData = (resolve, url, data) => (dispatch) => {
  dataFetcher(dispatch, {
    url: url,
    method: 'GET',
    onSuccess: setBestPicks,
    data,
    onFailure: () => setContentError(true),
  }, resolve);
}

export const setBestPicks = ({data}) => ({
  type: SET_BEST_PICKS,
  payload: {data}
});

export const sortBestPicksThunk = (criteria, direction) => (dispatch, getState) => {
  dispatch(sortBestPicks(criteria, direction));
  const selectedLocation = getSelectedLocation(getState());
  generateBestPicksAndFetch(dispatch, getState(), selectedLocation.id, criteria)
}

export const sortBestPicks = (criteria, direction) => ({
  type: SORT_BEST_PICKS,
  payload: {criteria, direction}
});

export const setInitalBestPicksCriteria = (criteria) => ({
  type: SET_INITIAL_BEST_PICKS_CRITERIA,
  payload: {criteria}
});
