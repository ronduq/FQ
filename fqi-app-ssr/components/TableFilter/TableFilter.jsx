import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Table from '../Table/Table';
import Tooltip from '../../hocs/TooltipContainer';
import NotFoundDisplay from '../NotFoundDisplay/NotFoundDisplay';
import get from 'lodash.get';

import InputTemplate from '../InputTemplate/InputTemplate';

import s from './TableFilter.scss';

class TableFilter extends React.Component {

  constructor(props){
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    const { producesData, tableInfo } = this.props;
    const indexProd = Object.keys(producesData.items).find(produce => !producesData.items[produce].isParent);
    const initProd = producesData.items[indexProd].id;
    if (Object.keys(tableInfo).length === 0)  {
      this.props.setProduceInfoTable(initProd, producesData.itemsLabels[initProd]);
    }
  }
  
  handleClick (produce) {
    const { updateProduce, producesData } = this.props;
    updateProduce(0, true, 'parentCode', false, produce);
    this.props.setProduceInfoTable(produce, producesData.itemsLabels[produce]);
  }

  handleChange (event) {
    const produce = event.target.value;
    this.handleClick(produce);
  }

  render (){
    const {
      producesData, 
      retailerRanksData, 
      retailersData,
      selectedLocation,
      sortRanks,
      viewport,
      tableInfo,
      notFoundScreens,
      retailerProduces,
      content,
    } = this.props;
    const highlight =  get(tableInfo,['produce'],'');
    const title =  get(tableInfo,['title'],'');

      return (
        <div className={s.container}>
          <div className={s.title}>
            <Tooltip 
              side='right' 
              size='large'
              bubblePosition='right'
              id='all_produce_title'
            >
              <h3>{content.table_title}</h3>
            </Tooltip>  
          </div>  
          <div className={s.body}>
            {viewport !== 'mobile' &&
            <ul className={s.filter} id={`tableFilterButton`}>
              { Object.keys(producesData.items).map((prod, index) => {
                const produce = producesData.items[prod];
                if (produce.selected && !produce.isParent) {
                  return (
                  <li id={`test${index}`} key={`tableFilterListItem-${index}`} className={classNames({[s.highlight] : produce.id === highlight })}>
                    <button className={s.filterButton} onClick={() => this.handleClick(produce.id)}>{producesData.itemsLabels[produce.id]}</button>
                  </li>
                  )
                }
              })}
            </ul> }
            { viewport === 'mobile' &&
              <div className={s.dropdownContainer}>
                <label className={s.dropdownLabel} htmlFor="produceList">{content.table_mobile_title}</label>
                <select className={s.dropdown} id="produceList" name="select" onChange={this.handleChange}>
                { Object.keys(producesData.items).map((prod, index) => {
                    const produce = producesData.items[prod];
                    if (produce.selected && !produce.isParent) {
                      return (
                      <option key={`tableFilterListItem-${index}`} value={produce.id}>
                        {producesData.itemsLabels[produce.id]}
                      </option>
                      )
                    }
                  })}
                </select> 
              </div>
            }  

            <div className={s.content}>
              <h3 className={s.contentTitle}>{title}</h3>
              { Object.keys(retailerProduces.items).filter(item => retailerProduces.items[item].produceCode === highlight).length > 0 &&
                <Table 
                  retailersData={retailersData} 
                  sortRanks={sortRanks} 
                  noBorders={true}
                  labels={retailerRanksData.labels}
                  selected={retailerRanksData.selected}
                  direction= {retailerRanksData.direction}
                  ranks={retailerRanksData.ranksSingle}
                  type='single'
                  location={selectedLocation}
                  produceId={highlight}
                />
              }
              { Object.keys(retailerProduces.items).filter(item => retailerProduces.items[item].produceCode === highlight).length == 0 &&
                <div className={s.notFoundDisplayContainer}>
                  <NotFoundDisplay 
                    type='notFoundVeg' 
                    title={notFoundScreens[`retailer_produce`].title}
                    text={notFoundScreens[`retailer_produce`].text} 
                  />
                </div>
              }
            </div>
          </div>
        </div>
      )}
};

TableFilter.propTypes = {
  producesData: PropTypes.shape(), 
  retailerRanksData: PropTypes.shape(), 
  retailersData: PropTypes.shape(), 
  sortRanks: PropTypes.func,
  selectedLocation: PropTypes.string,
  viewport: PropTypes.string,
  tableInfo: PropTypes.shape(),
};

TableFilter.defaultProps = {
  producesData: {}, 
  retailerRanksData: {}, 
  retailersData: {}, 
  sortRanks: () => {}, 
  viewport: '',
  tableInfo: {},
};

export default TableFilter;
