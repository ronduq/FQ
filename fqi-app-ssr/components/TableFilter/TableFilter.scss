@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.title {
  padding: 10px 15px;
  position: relative;
  z-index: 4;
  text-align: center;

  h3{
    @include font-family('proxima-bold');
    @include font-size('medium-20');
  }

  @include respond-to('tablet', min-width) {
    background: white;
    box-shadow: 0px 2px 3px 1px rgba(126, 126, 126, 0.3);
    text-align: left;

    h3{
      @include font-family('proxima-regular');
      @include font-size('medium-20');
      transform: translateY(-1px);
    }
  }
}

.buttonInfo {
  color: #000;
  float: right;
}

.body {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
  background: white;

  @include respond-to('tablet', min-width) {
    border: 1px solid $colour-grey-medium;
    border-top: 0;
  }
  @media all and (-ms-high-contrast:none) {
    flex-wrap: nowrap;
  }
}

.filter {
  background: $colour-grey-light;
  display: flex;
  flex-direction: column;
  flex-basis: 100%;
  flex: 1 1 0;
  border-right: 1px solid $colour-grey-medium;
  position: relative;
  z-index: 3;
  @media all and (-ms-high-contrast:none) {
    flex-basis: 20%;
  }
}
.content {
  background: white;
  display: flex;
  flex-direction: column;
  flex-basis: 100%;
  flex: 4 1 0;
  border-left: 1px solid $colour-grey-medium;
  @media all and (-ms-high-contrast:none) {
    flex-basis: 80%;
  }

}

.contentTitle {
  padding: 20px;
  @include respond-to('mobile') {
    display: none;
  }
}

.filterButton{
  width: 100%;
  color: $colour-secondary;
  cursor: pointer;
  text-transform: uppercase;
  @include font-family('futura');
  @include font-size('medium');
  text-align: left;
  padding: 10px 15px;
  background: $colour-grey-light;
  box-sizing: border-box;
  border-bottom: 1px solid $colour-grey-medium;
  position: relative;

  .highlight &{
    background: white;
    color: $colour-grey-dark;

    &:after {
      content: '';
      position: absolute;
      left: 0;
      top: 34px;
      width: 0;
      height: 0;
      border: 8px solid transparent;
      border-left-color: $colour-secondary;
      border-right: 0;
      margin-top: -20px;
      margin-right: -20px;
    }
  }


  &:hover {
    color: $colour-secondary;
    text-decoration: underline;
  }
}

.dropdownContainer {
  display: block;
  width: 100%;
  z-index: 9;
}
.dropdown {
  display: block;
  width: calc(100% - 20px);
  background-color: $white;
  background-image: url('data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2220%22%20height%3D%2220%22%20viewBox%3D%220%200%2018%2011%22%3E%3Cpolyline%20stroke%3D%22%23FE744A%22%20fill%3D%22none%22%20stroke-width%3D%222%22%20points%3D%221%2C1.5%209%2C9.5%2017%2C1.5%20%22%3E%3C%2Fpolyline%3E%3C%2Fsvg%3E');
  background-repeat: no-repeat, repeat;
  background-position: right 16px top 50%, 0 0;
  background-size: 20px 20px;
  border-radius: 0;
  padding: 10px;
  margin: 0 auto;
  @include font-family('proxima-regular');
  @include font-size('medium');
}

.dropdownLabel {
  display: block;
  margin: 10px;
}

.notFoundDisplayContainer {
  transform:translateY(-90px);
  margin-bottom: 50px;
}
  

