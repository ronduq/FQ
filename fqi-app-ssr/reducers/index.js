import { combineReducers } from 'redux';
import contentReducer from './content';
import retailersReducer from './retailers';
import producesReducer from './produces';
import locationsReducer from './locations';
import retailerProducesReducer from './retailerProduces';
import retailerRanksReducer from './retailerRanks';
import produceInfoReducer from './produceInfo';
import retailerTrendsReducer from './retailerTrends';
import produceTrendsReducer from './produceTrends';
import bestPicksReducer from './bestPicks';
import breadcrumbsReducer from './breadcrumbs';

const appReducer = () =>
  combineReducers({
    content: contentReducer,
    locations: locationsReducer,
    produces: producesReducer,
    retailers: retailersReducer,
    retailerRanks: retailerRanksReducer,
    retailerProduces: retailerProducesReducer,
    produceInfo: produceInfoReducer,
    retailerTrends: retailerTrendsReducer,
    produceTrends: produceTrendsReducer,
    bestPicks: bestPicksReducer,
    breadcrumbs: breadcrumbsReducer,
  });

// const rootReducer = history => (state, action) => {
//   if (action.type === 'LOGOUT') {
//     state = undefined;
//   }
//   return appReducer()(state, action);
// };

export default appReducer;
