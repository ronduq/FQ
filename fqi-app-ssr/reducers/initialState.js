export const initialRetailersState =  {
  name: 'Retailers',
  items: [],
  allItemsSelected: true,
  itemsLabels: {},
  itemsLogos: {},
  selectedRetailer: '',
}

export const initialProducesState =  {
  name: 'Produces',
  allItemsSelected: true,
  initialSelection: true,
  items: {},
  itemsImages: {},
  itemsLabels: {},
}

export const initialContentState = {
  contentFetched: false,
  footer: [],
  viewport: 'mobile',
  retailerTrends: {
    show_view_by_dropdown: false,
  },
  homepage: {},
}

export const initialLocationState = {
  locations: [],
  selectedLocation: {}
}

export const initialRetailerRanksState = {
  ranks: [],
  labels: {
    retailer: 'Retailer',
    quality: 'Quality Score',
    value: 'Value Rank',
    perception: 'Perception Rank',
    single_produce_average_title: 'Selected Produce Averagedddd'
  },
  ranksSingle: [],
  selected: 'quality',
  direction: 'topToBottom',
}

export const producesOfRetailer = {
  items: [],
  sortDirection: 'DSC',
  compareBy: 'quality',
  infoTable: {}
}

export const produceInfo = {
  info: {},
  labels: {},
}

export const initialRetailerTrendsState = {
  trends: [],
  viewBy: 'weekly',
  compareBy: 'quality',
  isLoading: false,
}

export const initialProduceTrendsState = {
  trends: [],
  viewBy: 'weekly',
  compareBy: 'quality',
  isLoading: false,
}

export const bestPicksState = {
  criteria: 'quality',
  topPicks: [],
  retailerTopPicks: []
}

export const breadcrumbs = {
  
}
