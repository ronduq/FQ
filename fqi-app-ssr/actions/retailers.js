import { 
  UPDATE_RETAILER, 
  SELECT_ALL_RETAILERS_CHECKBOX, 
  SELECT_ALL_RETAILERS,
  SET_SELECTED_RETAILER,
  SET_INITIAL_RETAILERS,
  SET_RETAILERS_CONTENT,
  SET_RETAILERS_DATA
} from '../constants';

import { getSelectedLocation, getProduceInfoTable } from '../selectors';
import { setContentError } from './content';
import dataFetcher from '../lib/dataFetcher';
import { generateRetailersRankQueryAndFetch, setRetailerRanks } from './retailerRanks';
import { generateBestPicksAndFetch } from './bestPicks';

export const updateRetailerThunk = (retailerId, check, parentCode, isParent, produce, retailer) => (dispatch, getState) => {
  dispatch(updateRetailer(retailerId, check));
  dispatch(selectAllRetailersCheckbox());
  const selectedLocation = getSelectedLocation(getState());
  generateRetailersRankQueryAndFetch(dispatch, getState(), selectedLocation.id, '', retailer)
}

export const updateRetailerProfileThunk = (retailerId, check) => (dispatch) => {
  dispatch(updateRetailer(retailerId, check));
  dispatch(selectAllRetailersCheckbox());
  dispatch(fetchRetailerProfileDataThunk());
}

export const selectAllRetailerProfileThunk = currentSelection => (dispatch) => {
  dispatch(selectAllRetailers(null, currentSelection));
  dispatch(fetchRetailerProfileDataThunk());
}

export const sortDirectionRetailerProfileThunk = (criteria, direction) => (dispatch) => {
  dispatch(setRetailerRanks(criteria, direction));
  dispatch(fetchRetailerProfileDataThunk());
}

export const fetchRetailerProfileDataThunk = () => (dispatch, getState) => {
  const state = getState();
  const selectedLocation = getSelectedLocation(state)
  const selectedProduce = getProduceInfoTable(state).produce;
  generateRetailersRankQueryAndFetch(dispatch, getState(), selectedLocation.id, selectedProduce)
}

export const updateRetailerTrendsThunk = (retailerId, check) => (dispatch) => {
  dispatch(updateRetailer(retailerId, check));
  dispatch(selectAllRetailersCheckbox());
}

export const updateBestPicksRetailerThunk = (retailerId, check, parentCode, isParent, criteria) => (dispatch, getState) => {
  dispatch(updateRetailer(retailerId, check));
  dispatch(selectAllRetailersCheckbox());
  const selectedLocation = getSelectedLocation(getState());
  generateBestPicksAndFetch(dispatch, getState(), selectedLocation.id, criteria)
}

export const updateRetailer = (retailerId, check) => ({
  type: UPDATE_RETAILER,
  payload: { retailerId, check }
})

export const selectAllRetailersCheckbox = () => ({
  type: SELECT_ALL_RETAILERS_CHECKBOX,
})

export const selectAllRetailersThunk = (status, changeLocation, currentSelection = null, retailer) => (dispatch, getState) => {
  dispatch(selectAllRetailers(status, currentSelection));
  if (!changeLocation) {
    const selectedLocation = getSelectedLocation(getState());
    generateRetailersRankQueryAndFetch(dispatch, getState(), selectedLocation.id, '' , retailer)
  }
}

export const selectAllBestPicksRetailersThunk = (status, changeLocation, currentSelection = null, criteria) => (dispatch, getState) => {
  dispatch(selectAllRetailers(status, currentSelection));

  if (!changeLocation) {
    const selectedLocation = getSelectedLocation(getState());
    generateBestPicksAndFetch(dispatch, getState(), selectedLocation.id, criteria)
  }
}

export const selectAllRetailers = (status, currentSelection) => ({
  type: SELECT_ALL_RETAILERS,
  payload: { status, currentSelection }
})

export const setRetailersContent = (retailers) => ({
  type: SET_RETAILERS_CONTENT,
  payload: { retailers }
});

      
export const setSelectedRetailer = selectedRetailer => ({
  type: SET_SELECTED_RETAILER,
  payload: { selectedRetailer }
});

export const fetchRetailersData = (resolve, location) => (dispatch) => {
  dataFetcher(dispatch, {
    url: `${process.env.BASE_URL}/api/v1/${location}/retailers`,
    method: 'GET',
    onSuccess: setRetailersData,
    onFailure: () => setContentError(true),
  }, resolve);
}

export const setRetailersData = ({data}) => ({
  type: SET_RETAILERS_DATA,
  payload: { data }
})

export const setInitialRetailers = retailers => ({
  type: SET_INITIAL_RETAILERS,
  payload: { retailers }
});
