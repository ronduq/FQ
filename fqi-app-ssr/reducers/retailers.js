import update from 'react-addons-update';
import { initialRetailersState } from './initialState';

import { 
  UPDATE_RETAILER, 
  SELECT_ALL_RETAILERS_CHECKBOX, 
  SELECT_ALL_RETAILERS,
  SET_SELECTED_RETAILER,
  SET_INITIAL_RETAILERS,
  SET_RETAILERS_CONTENT,
  SET_RETAILERS_DATA
} from '../constants';

const retailersReducer = (state = initialRetailersState, action) => {
  const { payload, type } = action;
  switch (type) {
    case SET_INITIAL_RETAILERS: {
      const { retailers } = payload;
      return {
        ...state,
        initialRetailers: retailers.split(',')
      }
    }
    case SET_RETAILERS_DATA: {
      const { data } = payload;
      const items = data;
      const { initialRetailers } = state;

      items.forEach(item => {
        let selected = true;
        if (initialRetailers) selected = initialRetailers.includes(item.id);
        item.selected = selected;
      });

      const allItemsSelected = data.filter(item => item.selected).length === data.length;

      return {
        ...state,
        items,
        initialRetailers: null,
        allItemsSelected,
      }
    }
    case UPDATE_RETAILER: {
      const { retailerId, check } = payload;
      return update(state, {
        items: {
          [retailerId]: {
            selected: {$set: check}
          }
        }
      })
    }
    case SELECT_ALL_RETAILERS_CHECKBOX: {
      let allRetailersAreSelected = true;
      Object.keys(state.items).map((retailer) => {
        if (!state.items[retailer].selected) {
          allRetailersAreSelected = false;
        }
      })
      return update(state, {
        allItemsSelected: {$set: allRetailersAreSelected}
      })
    }
    case SELECT_ALL_RETAILERS: {
      const { status, currentSelection } = payload;
      let allItemsSelected = !state.allItemsSelected;
      if (status) allItemsSelected = status;

      const updatedItems = state.items.map((retailer) => {
        retailer.selected = allItemsSelected;
        if (currentSelection == retailer.retailerCode) retailer.selected = true;
        return retailer;
      })

      return {
        ...state,
        items: updatedItems,
        allItemsSelected
      }
    }
    case SET_RETAILERS_CONTENT: {
      const { retailers } = payload;
      let itemsLabels = {};
      let itemsLogos ={};
      retailers.map((retailer) => {
        itemsLabels[retailer.id] = retailer.label;
        itemsLogos[retailer.id] = retailer.logo;
      });
      return update(state, {
        itemsLabels: {$set: itemsLabels},
        itemsLogos: {$set: itemsLogos},
      })
    }
    case SET_SELECTED_RETAILER: {
      const { selectedRetailer } = payload;
      return {
        ...state,
        selectedRetailer
      }
    }
    default:
      return state;
  }
};

export default retailersReducer;
