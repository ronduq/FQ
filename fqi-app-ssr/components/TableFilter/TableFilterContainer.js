import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import TableFilter from './TableFilter';
import { 
  getRetailersState,
  getProducesState,
  getRetailerRanksState,
  getViewport,
  getSelectedLocationId,
  getProduceInfoTable,
  getRetailerProduces,
  getNotFoundScreens,
  getRetailerProfileContent,
} from '../../selectors/index';
import { setViewport } from '../../actions/content';
import { updateProduceThunk } from '../../actions/producesActions';
import { sortDirectionRetailerProfileThunk  } from '../../actions/retailers';
import { setProduceInfoTable } from '../../actions/retailerProduces'

const mapStateToProps = state => {
  return {
    viewport: getViewport(state),
    retailersData: getRetailersState(state),
    producesData: getProducesState(state),
    retailerRanksData: getRetailerRanksState(state),
    selectedLocation: getSelectedLocationId(state),
    tableInfo: getProduceInfoTable(state),
    retailerProduces: getRetailerProduces(state),
    notFoundScreens: getNotFoundScreens(state),
    content: getRetailerProfileContent(state),
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setViewport: (width) => setViewport(width),
      sortRanks: (criteria, direction, type) => sortDirectionRetailerProfileThunk(criteria, direction, type),
      updateProduce: (produceId, check, parentCode, isParent, produce) => updateProduceThunk(produceId, check, parentCode, isParent, produce),
      setProduceInfoTable: (produce, title) => setProduceInfoTable(produce, title),
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TableFilter);