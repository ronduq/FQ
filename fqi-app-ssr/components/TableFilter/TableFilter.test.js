/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { initialRetailerRanksState, initialRetailersState, producesOfRetailer, initialProducesState } from '../../reducers/initialState'
import TableFilter from './TableFilter'
import { updateRetailer } from '../../actions/retailers'

const retailerProduces = {
  allItemsSelected: true,
  items: {
    '0': {
      id: "avocado",
      isChild: false,
      isParent: false,
      produceCode: "avocado",
      produceName: "Avocado",
      selected: true,
    },
    '1': {
      id: "banana",
      isChild: false,
      isParent: false,
      produceCode: "banana",
      produceName: "Banana",
      selected: true,
    },
  },
  itemsImages: {
    avocado: "Avocado",
    banana: 'Banana',
  },
  itemsLabels: {
    avocado: "Avocado",
    banana: 'Banana',
  }
}

const props = {
  retailersLabels:{
    single_produce_average_title: 'Single produce',
    quality: 'quality',
    value: 'value',
    perception: 'perception',
  },
  query: {
    location: 'boston',
    retailer: 'aldi'
  },
  selectedLocation: { label: 'Boston', id: 'boston'},
  selectedRetailer: 'aldi',
  
  retailersData: initialRetailersState,
  producesData: retailerProduces,
  cardLabels: {
    countries_of_origin: "Countries of origin",
    last_scan: "Last scan",
    organic: "Organic",
    perception: "Perception",
    product_of: "Product of",
    profile: "Profile",
    quality: "Quality",
    trends: "Trends",
    value: "Value",
  },
  retailerRanksData: initialRetailerRanksState,
  setProduceInfoTable: () => {}
};

const fakeStore = {
  dispatch: jest.fn(),
  getState: () => ({
    content: {
      tooltips:{
        trend:{}
      }
    }
  }),
  subscribe: jest.fn()
};

const updateProduce = jest.fn();



describe('With Snapshot Testing', () => {
  it('TableFilter snapshot"', () => {
    const component = renderer.create(<Provider store={fakeStore}><TableFilter {...props}/></Provider>)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

  it('should call updateProduce when button clicked', () => {
    const wrapper = mount(
      <Provider store={fakeStore}>
        <TableFilter {...props} updateProduce={updateProduce}/>
      </Provider>);
    const button = wrapper.find('#test0 button');
    button.simulate('click');
    expect(updateProduce).toHaveBeenCalledTimes(1);
  });

})