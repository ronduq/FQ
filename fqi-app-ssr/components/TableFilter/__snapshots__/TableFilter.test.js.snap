// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing TableFilter snapshot" 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <div
      className="wrapper"
    >
      <span
        className="content"
      >
        <h3>
          All Produce
        </h3>
      </span>
      
    </div>
  </div>
  <div
    className="body"
  >
    <ul
      className="filter"
      id="tableFilterButton"
    >
      <li
        className=""
        id="test0"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
          Avocado
        </button>
      </li>
      <li
        className=""
        id="test1"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
          Banana
        </button>
      </li>
    </ul>
    <div
      className="content"
    >
      <h3
        className="contentTitle"
      >
        
      </h3>
      <table
        aria-readonly="true"
        className="table noBorders"
        role="grid"
      >
        <caption>
          Retailers table sorted by 
          quality
          , 
          <span>
            ascending
          </span>
           
           
        </caption>
        <thead>
          <tr
            className="headerRow"
          >
            <th
              className="tableHeader cell"
              scope="col"
            >
              Retailer
            </th>
            <th
              className="tableHeader cell filterButton cellHighlight"
              id="tableHeaderQualityButton"
              scope="col"
            >
              <div
                className="wrapper"
              >
                
                <span
                  className="content"
                >
                  <button
                    onClick={[Function]}
                  >
                    Quality Score
                    <div
                      className="arrowsContainer"
                    >
                      <span
                        className="arrow arrowDown"
                      >
                        <svg
                          aria-hidden="true"
                          className="iconXSmall iconBlack"
                          focusable="false"
                        >
                          <use
                            xlinkHref="#chevron"
                          />
                        </svg>
                      </span>
                    </div>
                  </button>
                </span>
              </div>
            </th>
            <th
              className="tableHeader cell filterButton"
              id="tableHeaderValueButton"
              scope="col"
            >
              <div
                className="wrapper"
              >
                
                <span
                  className="content"
                >
                  <button
                    onClick={[Function]}
                  >
                    Value Rank
                    <div
                      className="arrowsContainer"
                    >
                      <span
                        className="arrow arrowUp"
                      >
                        <svg
                          aria-hidden="true"
                          className="iconXSmall iconBlack"
                          focusable="false"
                        >
                          <use
                            xlinkHref="#chevron"
                          />
                        </svg>
                      </span>
                      <span
                        className="arrow arrowDown"
                      >
                        <svg
                          aria-hidden="true"
                          className="iconXSmall iconBlack"
                          focusable="false"
                        >
                          <use
                            xlinkHref="#chevron"
                          />
                        </svg>
                      </span>
                    </div>
                  </button>
                </span>
              </div>
            </th>
            <th
              className="tableHeader cell filterButton"
              id="tableHeaderPerceptionButton"
              scope="col"
            >
              <div
                className="wrapper"
              >
                
                <span
                  className="content"
                >
                  <button
                    onClick={[Function]}
                  >
                    Perception Rank
                    <div
                      className="arrowsContainer"
                    >
                      <span
                        className="arrow arrowUp"
                      >
                        <svg
                          aria-hidden="true"
                          className="iconXSmall iconBlack"
                          focusable="false"
                        >
                          <use
                            xlinkHref="#chevron"
                          />
                        </svg>
                      </span>
                      <span
                        className="arrow arrowDown"
                      >
                        <svg
                          aria-hidden="true"
                          className="iconXSmall iconBlack"
                          focusable="false"
                        >
                          <use
                            xlinkHref="#chevron"
                          />
                        </svg>
                      </span>
                    </div>
                  </button>
                </span>
              </div>
            </th>
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
